pandas/core/generic.py

```diff
@@ -3151,150 +3167,338 @@ def to_latex(
+           internal `LatexFormatter` class. This is the reason for many of the
+           altered arguments in the method signature.
+
+           The Styler implementation has extended functionality and performance and
+           further information is available in its own documentation relevant for this
+           method.
+
         Examples
         --------
-        >>> df = pd.DataFrame(dict(name=['Raphael', 'Donatello'],
-        ...                   mask=['red', 'purple'],
-        ...                   weapon=['sai', 'bo staff']))
-        >>> print(df.to_latex(index=False))  # doctest: +NORMALIZE_WHITESPACE
+        >>> df = pd.DataFrame({{'name': ['Raphael', 'Donatello'],
+        ...                    'mask': ['red', 'purple'],
+        ...                    'weapon': ['sai', 'bo staff']}})
+        >>> print(df.to_latex(index=False, hrules=True))
         \begin{{tabular}}{{lll}}
-         \toprule
-               name &    mask &    weapon \\
-         \midrule
-            Raphael &     red &       sai \\
-          Donatello &  purple &  bo staff \\
+        \toprule
+        name & mask & weapon \\
+        \midrule
+        Raphael & red & sai \\
+        Donatello & purple & bo staff \\
         \bottomrule
         \end{{tabular}}
         """
-        # Get defaults from the pandas config
-        if self.ndim == 1:
-            self = self.to_frame()
-        if longtable is None:
-            longtable = config.get_option("display.latex.longtable")
-        if escape is None:
-            escape = config.get_option("display.latex.escape")
-        if multicolumn is None:
-            multicolumn = config.get_option("display.latex.multicolumn")
-        if multicolumn_format is None:
-            multicolumn_format = config.get_option("display.latex.multicolumn_format")
-        if multirow is None:
-            multirow = config.get_option("display.latex.multirow")
-
-        self = cast("DataFrame", self)
-        formatter = DataFrameFormatter(
-            self,
-            columns=columns,
-            col_space=col_space,
+        kwargs = locals()
+        deprecated_args = {
+            "longtable": '`environment="longtable"`',
+            "multicolumn": "`multicol_align`",
+            "multicolumn_format": "`multicol_align`",
+            "multirow": "`multirow_align`",
+            "col_space": None,
+            "formatters": "`formatter`, `precision`, `decimal`, `thousands`",
+            "float_format": "`precision`, `decimal`, `thousands`, `formatter`",
+            "sparsify": "`sparse_index`, `sparse_columns`",
+            "bold_rows": "`bold_header`",
+        }
+        for k, v in deprecated_args.items():
+            if kwargs[k] is not None:
+                warnings.warn(
+                    f"`{k}` is deprecated after transition to `Styler.to_latex()` "
+                    f"signature." + ("" if v is None else f" Consider {v} instead."),
```

Comment:
I suggest that this message is a bit refactored by extracting ``msg`` and ``recommendation``, so that it is slightly more readable.

``` python
msg = (
    f"`{k}` is deprecated "
    "after transition to `Styler.to_latex()` signature."
)
recommendation = f"Consider {v} instead" if v else ""
warnings.warn(
    ' '.join([msg, recommendation]),
    FutureWarning,
    stacklevel=2,
)
```